xarray/core/dataset.py

```diff
@@ -1582,6 +1582,17 @@ def reindex(self, indexers=None, method=None, tolerance=None, copy=True,
         indexers = utils.combine_pos_and_kw_args(indexers, kw_indexers,
                                                  'reindex')
 
+        from .dataarray import DataArray
+        import warnings
+        for dim, ind in indexers.items():
+            if isinstance(ind, DataArray) and ind.dims != ('dims', ):
+                warnings.warn(
```

Comment:
Can we put this warning in `alignment.reindex_variables` instead? That's where all the logic happens.

Reply:
Yes. Done.